css-view-transitions-2/Overview.bs

```diff
@@ -0,0 +1,400 @@
+
+## Setting up and activating the cross-document view transition ##
+
+	<div algorithm>
+		To <dfn>setup outbound cross-document view transition</dfn> given a {{Document}} |oldDocument|,
+		a {{Document}} |newDocument|, and |onReady|, which is an algorithm accepting nothing:
+
+		1. If |oldDocument|'s [=environment settings object/origin=] is not [=same origin=] as
+			|newDocument|'s [=environment settings object/origin=] then call |onReady| and return.
+
+		1. If |newDocument| [=was created via cross-origin redirects=] is true
+			and |newDocument|'s [=latest entry=] is null, then call |onReady| and return.
+
+			Note: A document with a non-null [=latest entry=]
+			is being [=Document/reactivated=], in which case we don't need to check for cross-origin redirects.
+
+		1. If |oldDocument| does not [=opt in to cross-document view transitions=], then call |onReady| and return.
+
+			Note: We don't know yet if |newDocument| has opted in, as it might not be parsed yet.
+			We check the opt-in for |newDocument| when it is [=reveal document|revealed=].
+
+		1. If |oldDocument|'s [=active view transition=] is not null,
+			then [=skip the view transition|skip=] |oldDocument|'s [=active view transition=]
+			with an "{{AbortError}}" {{DOMException}} in |oldDocument|'s [=relevant Realm=].
+
+			Note: this means that any running transition would be skipped when the document is ready
+			to unload.
+
+		1. Let |outboundTransition| be a new {{ViewTransition}} object in |oldDocument|'s [=relevant Realm=],
+			whose [=ViewTransition/process old state captured=] is set to the following steps:
+
+			1. |outboundTransition|'s [=ViewTransition/phase=] is not "`pending-capture`", then call |onReady| and return.
+
+			1. If |oldDocument| does not [=opt in to cross-document view transitions|opts in to cross-document view transitions=],
+				then [=queue a global task=] on the [=DOM manipulation task source=] given |newDocument|'s [=relevant global object=],
+				to perform the following step:
+
+				1. Let |newDocument|'s [=active view transition=] be a new {{ViewTransition}} in |newDocument|'s [=relevant Realm=],
+					whose [=ViewTransition/named elements=] is |outboundTransition|'s [=ViewTransition/named elements=],
+					[=ViewTransition/initial snapshot containing block size=] is |outboundTransition|'s [=ViewTransition/initial snapshot containing block size=],
+					and whose [=ViewTransition/is cross-document=] is true.
+
+				1. [=Call the update callback=] for |newDocument|'s [=active view transition=].
+
+				1. Call |onReady|.
+
+			Note: |outboundTransition| is not exposed to JavaScript, it is used only for capturing
+			the state of the old document.
+
+		1. Set |oldDocument|'s [=active view transition=] to |outboundTransition|.
+	</div>
+
+	<div algorithm>
+		To get the <dfn>inbound cross-document view-transition</dfn> for a {{Document}} |document|:
+
+		1. If |document| does not [=opt in to cross-document view transitions=], then return null.
```

Comment:
I don't think we should have a check for this here. If there is an "active view transition" whose "is cross document" is true then that's always returned. The places where the opt in is checked should be explicit and we should clear the "active view transition" at those places if needed.